lib/util.py

```diff
@@ -355,7 +355,17 @@ def user_dir():
     if 'ANDROID_DATA' in os.environ:
         return android_check_data_dir()
     elif os.name == 'posix':
-        return os.path.join(os.environ["HOME"], ".electron-cash" )
+        # Prefer XDG Base directory spec based condiguration path, but use the legacy dot-directory
+        # configuration path if user has upgraded from a previous installation
+        xdg_config_home = os.environ.get("XDG_CONFIG_HOME", os.path.join(os.environ["HOME"], ".config"))
+        config_path_modern = os.path.join(xdg_config_home, "electron-cash")
+        config_path_legacy = os.path.join(os.environ["HOME"], ".electron-cash")
+        if os.path.exists(config_path_modern) and os.listdir(config_path_modern):
+            return config_path_modern
```

Comment:
> Is this trying to handle a case where the user has both ~/.electron-cash and ~/.config/electron-cash?

Yes, it first probe for the new directory and if that doesn't exist (or is empty), it will probe for the old directory. If neither exist will will finally default to the new one. This is a pragmatic way to deal with the different possible locations that is used by several other software projects as well.

The alternative would be, of course, to always use the new directory and, if that isn't initialized yet, try to move the data in from the old location. This would have to be handled like the current *wallet upgrade* migrations through IMHO, since it breaks older versions after being completed. If you prefer it this way I'd be very thankful if you could provide some pointers as to were this should be implemented in the source code.